### PR TITLE
feat: use sub:username identity claim by default when persisting behind a feature flag

### PR DESCRIPTION
**Pull request by danielleadams**
#### Description of changes

This pull requests does the following:
- persists "sub::username" format on Mutation requests
- adds a feature flag that is default for existing apps to use "username"
- adds a warning for current owner based auth on compile either explicitly use "username" if they want to keep using it 

#### Description of how you validated changes

- yarn test
- e2e tests in a later PR
- manual testing matrix

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by akshbhu — APPROVED** [review]
Some minor comments but overall LGTM

**Review by marcvberg — COMMENTED** [review]
Looks like some snapshots need to be updated, otherwise seems good

File: packages/amplify-e2e-tests/src/__tests__/api_2.test.ts [hunk 1]
```diff
@@ -269,17 +272,47 @@ describe('amplify add api (GraphQL)', () => {
...
+    const transformConfigWithDS = getTransformConfig(projRoot, name);
     expect(transformConfigWithDS).toBeDefined();
     expect(transformConfigWithDS.ResolverConfig).toBeDefined();
     expect(transformConfigWithDS.ResolverConfig.project).toBeDefined();
     expect(transformConfigWithDS.ResolverConfig.project.ConflictHandler).toEqual('AUTOMERGE');
     expect(transformConfigWithDS.ResolverConfig.project.ConflictDetection).toEqual('VERSION');
   });
 
-  it('init a project and add custom iam roles - local test with gql v2', async () => {
+  it.only('init a project and add custom iam roles - local test with gql v2', async () => {
```
Comment (danielleadams) [inline packages/amplify-e2e-tests/src/__tests__/api_2.test.ts:283]: remove this

File: .eslint-dictionary.json [hunk 2]
```diff
@@ -37,9 +37,9 @@
   "globals",
   "graphql",
   "homedir",
-  "integtestFn",
   "iam",
   "indicies",
+  "integtestFn",
```
[the remark this answers is not in the document]
Comment (danielleadams) [reply .eslint-dictionary.json:45]: It was reordered, not added, so I'll leave it and we can remove non words in another PR.

**Comment by danielleadams** [comment]
@marcvberg this has been rebased

**Review by marcvberg — APPROVED** [review]
LGTM

**Comment by danielleadams** [comment]
When https://github.com/aws-amplify/amplify-cli/pull/10222 is merged, these tests will pass (they are due to mismatched snapshots)

**Pull request merged**